src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Tcp/IdentityTests.cs

```diff
@@ -1,43 +1,75 @@
 ﻿using System;
+using System.Collections;
 using System.ServiceModel;
+using System.ServiceModel.Security;
 using System.Text;
 using Xunit;
 
-namespace Security.TransportSecurity.Tests.Tcp
+using Infrastructure.Common;
+
+public static class IdentityTests
 {
-    public static class IdentityTests
+    [Fact]
+    [OuterLoop]
+    // The product code will check the Dns identity from the server and throw if it does not match what is specified in DnsEndpointIdentity
+    public static void VerifyServiceIdentityMatchDnsEndpointIdentity()
     {
-        [Fact]
-        [ActiveIssue(12)]
-        [OuterLoop]
-        // The product code will check the Dns identity from the server and throw if it does not match what is specified in DnsEndpointIdentity
-        public static void VerifyServiceIdentityMatchDnsEndpointIdentity()
+        string testString = "Hello";
+
+        NetTcpBinding binding = new NetTcpBinding();
+        binding.Security.Mode = SecurityMode.Transport;
+        binding.Security.Transport.ClientCredentialType = TcpClientCredentialType.None;
+
+        EndpointAddress endpointAddress = new EndpointAddress(new Uri(Endpoints.Tcp_VerifyDNS_Address), new DnsEndpointIdentity(Endpoints.Tcp_VerifyDNS_HostName));
+        ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, endpointAddress);
+        IWcfService serviceProxy = factory.CreateChannel();
+
+        try
         {
-            string testString = "Hello";
+            var result = serviceProxy.Echo(testString);
+            Assert.Equal(testString, result);
+        }
+        finally
+        {
```

Comment:
Nit: should probably use the close pattern Stephen has been using (do the close before the finally, and then have the finally call the testhelper to close any remaining CommunicationObjects